packages/scope-manager/src/scope/ScopeBase.ts

```diff
@@ -87,22 +87,20 @@ function isStrictScope(
     if (stmt.type !== AST_NODE_TYPES.ExpressionStatement) {
       break;
     }
-    const expr = stmt.expression;
 
-    if (
-      expr.type !== AST_NODE_TYPES.Literal ||
-      typeof expr.value !== 'string'
-    ) {
+    if (stmt.directive === 'use strict') {
+      return true;
+    }
+
+    const expr = stmt.expression;
+    if (expr.type !== AST_NODE_TYPES.Literal) {
       break;
     }
-    if (expr.raw !== null && expr.raw !== undefined) {
-      if (expr.raw === '"use strict"' || expr.raw === "'use strict'") {
-        return true;
-      }
-    } else {
-      if (expr.value === 'use strict') {
-        return true;
-      }
+    if (expr.raw === '"use strict"' || expr.raw === "'use strict'") {
+      return true;
+    }
+    if (expr.value === 'use strict') {
+      return true;
```

Comment:
those checks are redundant, as we already checking directive field

Reply:
these checks are just legacy things from the eslint-scope. I assume they existed for backwards compat reasons